it/uk/gov/hmrc/minorentityidentificationfrontend/controllers/uaControllers/CaptureOfficePostcodeControllerISpec.scala

```diff
@@ -0,0 +1,126 @@
+      testCaptureOfficePostcodeView(result)
+    }
+
+    "redirect to sign in page" when {
+      "the user is UNAUTHORISED" in {
+        stubAuthFailure()
+        lazy val result: WSResponse = get(s"/identify-your-unincorporated-association/$testJourneyId/registered-office-postcode")
+
+        result must have(
+          httpStatus(SEE_OTHER),
+          redirectUri("/bas-gateway/sign-in" +
+            s"?continue_url=%2Fidentify-your-unincorporated-association%2F$testJourneyId%2Fregistered-office-postcode" +
+            "&origin=minor-entity-identification-frontend"
+          )
+        )
+      }
+    }
+
+  }
+
+  "POST /registered-office-postcode" when {
+    "the registered office postcode is correctly formatted" should {
+      "redirect to CYA page" in {
+        await(insertJourneyConfig(
+          journeyId = testJourneyId,
+          internalId = testInternalId,
+          testUnincorporatedAssociationJourneyConfig(businessVerificationCheck = true)
+        ))
+        enable(EnableFullUAJourney)
+        stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
+        stubStoreOfficePostcode(testJourneyId, testOfficePostcode)(status = OK)
+
+        lazy val result = post(s"/identify-your-unincorporated-association/$testJourneyId/registered-office-postcode")("officePostcode" -> testOfficePostcode)
+
+        result.status mustBe NOT_IMPLEMENTED
+        //TODO: Update this to redirect to CYA page when ready
+      }
+      "no registered office postcode is submitted" should {
+        lazy val result = {
+          await(insertJourneyConfig(
+            journeyId = testJourneyId,
+            internalId = testInternalId,
+            testUnincorporatedAssociationJourneyConfig(businessVerificationCheck = true)
+          ))
+          enable(EnableFullUAJourney)
+          stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
+          post(s"/identify-your-unincorporated-association/$testJourneyId/registered-office-postcode")("officePostcode" -> "")
+        }
+
+        "return a bad request" in {
+          result.status mustBe BAD_REQUEST
+        }
+
+        testCaptureOfficePostcodeErrorMessageNoEntryPostcode(result)
+      }
+
+      "an invalid registered office postcode is submitted" should {
+        lazy val result = {
+          await(insertJourneyConfig(
+            journeyId = testJourneyId,
+            internalId = testInternalId,
+            testUnincorporatedAssociationJourneyConfig(businessVerificationCheck = true)
+          ))
+          enable(EnableFullUAJourney)
+          stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
+          post(s"/identify-your-unincorporated-association/$testJourneyId/registered-office-postcode")("officePostcode" -> "AA!0!!")
+        }
+
+        "return a bad request" in {
+          result.status mustBe BAD_REQUEST
+        }
+
+        testCaptureOfficePostcodeErrorMessageInvalidPostcode(result)
+      }
+    }
+  }
+}
```

Comment:
I would add a test for a post by an unauthorised user. It wouldn't be easy to submit such a request, but it is good to ensure all request endpoints are protected.

Reply:
done